api/spec/src/billing/invoices/codes.tsp

```diff
@@ -0,0 +1,37 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/openapi3";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.OpenAPI;
+
+namespace Invoices;
+
+@pattern("^[A-Za-z0-9]+([\\.\\-\\/ _]?[A-Za-z0-9]+)*$")
+@minLength(1)
+@maxLength(32)
+@friendlyName("BillingCode")
+scalar Code extends string;
+
+// TODO: This is an enum, coming from code.json, we should use the json schema to have the available values
+@minLength(3)
+@maxLength(3)
+@friendlyName("BillingCurrencyCode")
+scalar CurrencyCode extends string;
```

Comment:
We already have currency code:
https://github.com/openmeterio/openmeter/blob/main/api/spec/src/types.tsp#L229-L238